libs/datamodel/parser-database/src/walkers/model/primary_key.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{
+    ast,
+    types::IdAttribute,
+    walkers::{ModelWalker, ScalarFieldAttributeWalker, ScalarFieldWalker},
+    ParserDatabase,
+};
+
+#[derive(Copy, Clone)]
+pub struct PrimaryKeyWalker<'ast, 'db> {
+    pub(crate) model_id: ast::ModelId,
+    pub(crate) attribute: &'db IdAttribute<'ast>,
+    pub(crate) db: &'db ParserDatabase<'ast>,
+}
+
+impl<'ast, 'db> PrimaryKeyWalker<'ast, 'db> {
```

Comment:
We should aim for good rustdocs here. Like top notch stuff...

But in later PRs...

Reply:
Next 7am-3pm non stop PR wil lbe docs, promised.